internal/backends/local/local.go

```diff
@@ -0,0 +1,157 @@
+package local
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+	"sync"
+
+	"github.com/cloudquery/plugin-sdk/specs"
+)
+
+type Local struct {
+	sourceName string
+	spec       Spec
+	tables     map[string]entries // table -> key -> value
+	tablesLock sync.RWMutex
+}
+
+type entries map[string]string
+
+func New(sourceSpec specs.Source) (*Local, error) {
+	spec := Spec{}
+	err := sourceSpec.UnmarshalBackendSpec(&spec)
+	if err != nil {
+		return nil, err
+	}
+	spec.SetDefaults()
+
+	l := &Local{
+		sourceName: sourceSpec.Name,
+		spec:       spec,
+	}
+	tables, err := l.loadPreviousState()
+	if err != nil {
+		return nil, err
+	}
+	if tables == nil {
+		tables = map[string]entries{}
+	}
+	l.tables = tables
+	return l, nil
+}
+
+func (l *Local) loadPreviousState() (map[string]entries, error) {
+	files, err := os.ReadDir(l.spec.Path)
+	if os.IsNotExist(err) {
+		return nil, nil
+	}
+	var tables = map[string]entries{}
+	for _, f := range files {
+		if f.IsDir() || !f.Type().IsRegular() {
+			continue
+		}
+		name := f.Name()
+		if !strings.HasSuffix(name, ".json") || !strings.HasPrefix(name, l.sourceName+"-") {
+			continue
+		}
+		table, kv, err := l.readFile(name)
+		if err != nil {
+			return nil, err
+		}
+		tables[table] = kv
+	}
+	return tables, nil
+}
+
+func (l *Local) readFile(name string) (table string, kv entries, err error) {
+	p := path.Join(l.spec.Path, name)
+	f, err := os.Open(p)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to open state file %v: %w", name, err)
```

Comment:
stdlib functions used in this method all wrap their errors with `os.PathError` (99% of the time) so the extra wrapping is unnecessary.

Reply:
I'll remove the name of the file but keep some wrapping so we can identify where the error happened